core/src/apps/monero/xmr/mlsag.py

```diff
@@ -273,6 +277,158 @@ def generate_mlsag(message, pk, xx, kLRki, index, dsRows, mg_buff):
     mg_buff[-1] = crypto.encodeint(cc)
 
 
+def generate_clsag_simple(message, pubs, in_sk, a, cout, index, mg_buff):
+    """
+    CLSAG for RctType.Simple
+    https://eprint.iacr.org/2019/654.pdf
+
+    Corresponds to proveRctCLSAGSimple in rctSigs.cpp
+
+    :param message: the full message to be signed (actually its hash)
+    :param pubs: vector of MoneroRctKey; this forms the ring; point values in encoded form; (dest, mask) = (P, C)
+    :param in_sk: CtKey; spending private key with input commitment mask (original); better_name: input_secret_key
+    :param a: mask from the pseudo output commitment; better name: pseudo_out_alpha
+    :param cout: pseudo output commitment; point, decoded; better name: pseudo_out_c
+    :param kLRki: used only in multisig, currently not implemented
+    :param index: specifies corresponding public key to the `in_sk` in the pubs array
+    :param mg_buff: buffer to store the signature to
+    """
+    cols = len(pubs)
+    if cols == 0:
+        raise ValueError("Empty pubs")
+
```

Comment:
You should check `cols == len(cout)`.

Reply:
I am adding the typing info, cout is Ge25519